index/dlg_index.c

```diff
@@ -1338,8 +1341,16 @@ struct Mailbox *mutt_index_menu(struct MuttWindow *dlg, struct Mailbox *m_init)
     nm_db_debug_check(shared->mailbox);
 #endif
 
-    int rc = index_function_dispatcher(priv->win_index, op);
+    int rc = IR_UNKNOWN;
 
+#ifdef USE_SIDEBAR
+    struct MuttWindow *win_sidebar = window_find_child(dlg, WT_SIDEBAR);
+    rc = sb_function_dispatcher(win_sidebar, op);
+    if ((rc != IR_ERROR) && (rc != IR_UNKNOWN))
```

Comment:
If you continue on `IR_ERROR` you'll get a `Key is not found` on, e.g., `unmailboxes *; sidebar-first`.

Reply:
On the other hand, if you don't, you won't be able to quit the `messages` window anymore.. doh..